### ISSUE TITLE
Unable to connect to docker cluster in tutorial

### ISSUE THREAD

**Issue by brianhks** [YugaByte/yugabyte-db#5]
As per the documentation it says I can use localhost:7000 and localhost:9000 to connect to the admin UI, this doesn't seem to work.  From the startup commands I do not see where any ports are "exposed"

>docker run --name yb-master-n1 --privileged --net yb-net --detach yugabytedb/yugabyte:latest /home/yugabyte/bin/yb-master --fs_data_dirs=/mnt/disk0,/mnt/disk1 --master_addresses=yb-master-n1:7100,yb-master-n2:7100,yb-master-n3:7100 --rpc_bind_addresses=yb-master-n1:7100
Adding node yb-master-n1
docker run --name yb-master-n2 --privileged --net yb-net --detach yugabytedb/yugabyte:latest /home/yugabyte/bin/yb-master --fs_data_dirs=/mnt/disk0,/mnt/disk1 --master_addresses=yb-master-n1:7100,yb-master-n2:7100,yb-master-n3:7100 --rpc_bind_addresses=yb-master-n2:7100
Adding node yb-master-n2
docker run --name yb-master-n3 --privileged --net yb-net --detach yugabytedb/yugabyte:latest /home/yugabyte/bin/yb-master --fs_data_dirs=/mnt/disk0,/mnt/disk1 --master_addresses=yb-master-n1:7100,yb-master-n2:7100,yb-master-n3:7100 --rpc_bind_addresses=yb-master-n3:7100
Adding node yb-master-n3
docker run --name yb-tserver-n1 --privileged --net yb-net --detach yugabytedb/yugabyte:latest /home/yugabyte/bin/yb-tserver --fs_data_dirs=/mnt/disk0,/mnt/disk1 --tserver_master_addrs=yb-master-n1:7100,yb-master-n2:7100,yb-master-n3:7100 --rpc_bind_addresses=yb-tserver-n1:9100
Adding node yb-tserver-n1
docker run --name yb-tserver-n2 --privileged --net yb-net --detach yugabytedb/yugabyte:latest /home/yugabyte/bin/yb-tserver --fs_data_dirs=/mnt/disk0,/mnt/disk1 --tserver_master_addrs=yb-master-n1:7100,yb-master-n2:7100,yb-master-n3:7100 --rpc_bind_addresses=yb-tserver-n2:9100
Adding node yb-tserver-n2
docker run --name yb-tserver-n3 --privileged --net yb-net --detach yugabytedb/yugabyte:latest /home/yugabyte/bin/yb-tserver --fs_data_dirs=/mnt/disk0,/mnt/disk1 --tserver_master_addrs=yb-master-n1:7100,yb-master-n2:7100,yb-master-n3:7100 --rpc_bind_addresses=yb-tserver-n3:9100
Adding node yb-tserver-n3

In addition I'm trying to start a C* application in docker connected to the same network and I'm not able to connect to any of the nodes.  Do you by chance have the cassandra ports only listening on localhost?

**Comment by brianhks** [comment]
Here is the error from my client:
Error in custom provider, com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: yb-master-n2/172.21.0.3:9042 (com.datastax.driver.core.exceptions.TransportException: [yb-master-n2/172.21.0.3:9042] Cannot connect), yb-master-n1/172.21.0.2:9042 (com.datastax.driver.core.exceptions.TransportException: [yb-master-n1/172.21.0.2:9042] Cannot connect), yb-master-n3/172.21.0.4:9042 (com.datastax.driver.core.exceptions.TransportException: [yb-master-n3/172.21.0.4:9042] Cannot connect))

**Comment by brianhks** [comment]
Also by way of suggestion, it is great that you can add and remove nodes from the cluster so nicely but as a first time user that isn't what I care about.  I want to see if my application will work with your database.  It would be nice if the examples showed how to start a single node yugabyte server and then connect an external application to it.  If my application doesn't work with your database it doesn't matter how many nodes I can add to the cluster.

**Comment by ramkumarvs** [comment]
@brianhks agree totally, And that is how we have laid out our documentation as well, first you start with creating a rf=3 cluster and connect cqlsh, redis client and run sample apps before we talk about fault tolerance and auto rebalance. That said, let me investigate the issue you are running into.

**Comment by kmuthukk** [comment]
@brianhks:

For yb-master & yb-tserver, 7000 and 9000 are the defaults in code.

# master
src/yb/master/master_defaults.h:constexpr uint16_t kMasterDefaultWebPort = 7000;
src/yb/master/master_main.cc:  FLAGS_webserver_port = kMasterDefaultWebPort;

# tserver
src/yb/tserver/tablet_server.h:  static const uint16_t kDefaultWebPort = 9000;
src/yb/tserver/tablet_server_main.cc:  FLAGS_webserver_port = TabletServer::kDefaultWebPort;

To override, you must explicitly pass/set the gflag, webserver_port. 

That said, your question as to why localhost:7000 is not working in the docker setup still remains. Will get back to you on this. [CC:  @ramkumarvs or @rkarthik007]

**Comment by ramkumarvs** [comment]
@brianhks yes that is if you don't have cqlsh running locally on your machine, we already ship one in the container for you to connect to. If you want to connect your app to YugaByte db, you pick the tserver container IP and use that. 
If you run 
`./yb-docker-ctl status`
you will see the master and tserver ip's along with the respective UI access information. 

For you to connect your app running locally to yugabyte DB, pick any one of the tserver node ips and you should be able to connect to it via cassandra port (9042).
ex.
`java -jar yb-sample-apps.jar --workload CassandraKeyValue --nodes 172.18.0.5:9042`
in my case  172.18.0.5 was my tserver-n1's docker container ip

**Comment by brianhks** [comment]
OK I'm an idiot.  I didn't know the host could talk directly to the container IP addresses, I do now.  I was able to connect my application.

It would be nice if the documentation called this out better instead of trying to connect to localhost:7000 for the UI or export those ports from the container.
